internal/db/client.go

```diff
@@ -91,11 +91,51 @@ func (c *Client) Get(inode uint64) (*common.Metadata, error) {
 	return c.parseRow(row)
 }
 
+func (c *Client) GetChildren(inode uint64) ([]common.Metadata, error) {
+	query, err := c.db.Prepare("SELECT * FROM files WHERE parent=?")
+	if err != nil {
+		return nil, fmt.Errorf("couldn't prepare statement: %v", err)
+	}
+
+	row, err := query.Query(inode)
+	if err != nil {
+		return nil, fmt.Errorf("there is an error in query: %v", err)
+	}
+	defer row.Close()
+
+	mdList := []common.Metadata{}
+	for row.Next() {
+		md, err := c.parseCurrentRow(row)
+		if err != nil {
+			return nil, err
```

Comment:
Nit
```suggestion
			return nil, fmt.Errorf("could not parse row: %v", err);
```

Reply:
Sorry I didn't see this before the merge. Instead of this line, I think we should just change error in below line 
https://github.com/paddlesteamer/hdn-drv/blob/5a9688692e43fe789f6749cdd667f788e53bd6de/internal/db/client.go#L128
to 'could not parse row'

This one too:
https://github.com/paddlesteamer/hdn-drv/blob/5a9688692e43fe789f6749cdd667f788e53bd6de/internal/db/client.go#L143

Reply:
@paddlesteamer No worries, I commented after the PR got merged :)

> Instead of this line, I think we should just change error in below line

I think instead of returning the `err` as it's received from `parseCurrentRow`, we'd better wrap the error on every level so that it's easier to trace later.

Reply:
Then we should be more descriptive. I did it like that because it it just returns an error from internal function. Wrapping 'could not parse' error with 'could not parse' error again will be just verbose

Reply:
@paddlesteamer BTW, `parseRow` and `parseCurrentRow` seems to be duplicates of each other.

Reply:
Arguments are different but the function bodies are identical.

Reply:
I know, do you have a suggestion on this? I couldn't pass sql.rows as sql.row. Maybe we can use a union for this?

Reply:
Ok, I found. Changing this line
https://github.com/paddlesteamer/hdn-drv/blob/5a9688692e43fe789f6749cdd667f788e53bd6de/internal/db/client.go#L89

to
```go
row := query.QueryRow(inode)
row.Next() 
```
may work. Then I can use parseCurrentRow for everything.

Reply:
@paddlesteamer Nice :+1: